docs/src/use/configure/migration-guide.md

```diff
@@ -0,0 +1,422 @@
+```
+
+You can add each of these configs to the exported array or expose specific rules from them.
+
+You must import the modules for local config files and npm package configs with `eslint.config.js`:
+
+```javascript
+// eslint.config.js
+
+import js from "@eslint/js";
+import customConfig from "./custom-config.js";
+import myConfig from "eslint-config-my-config";
+
+export default [
+    js.configs.recommended,
+    customConfig,
+    myConfig,
+    {
+        rules: {
+            semi: ["warn", "always"]
+        },
+        // ...other config
+    }
+];
+```
+
+Note that because you are just importing JavaScript modules, you can mutate the config objects before exporting them. For example, you might want to have a certain config object only apply to your test files:
+
+```javascript
+// eslint.config.js
+
+import js from "@eslint/js";
+import customTestConfig from "./custom-test-config.js";
+
+export default [
+    js.configs.recommended,
+    {
+        ...customTestConfig,
+        files: ["**/*.test.js"],
+    },
+];
+```
+
+### Ignoring Files
+
+#### .eslintrc
+
+To make ESLint ignore files with `.eslintrc`, you can create a separate `.eslintignore` file in the root of your project. The `.eslintignore` file uses the same glob pattern syntax as `.gitignore` files.
+
+```shell
+# .eslintignore
+temp.js
+.config/*
+# ...other ignored files
+```
+
+Alternatively, you can use an `ignorePatterns` property in your `.eslintrc` file:
+
+```javascript
+// .eslintrc.js
+{
+    // ...other config
+    ignorePatterns: ["temp.js", ".config/*"],
+}
+```
+
+#### eslint.config.js
+
+To ignore files with `eslint.config.js`, you can use the `ignores` property in a config object:
+
+```javascript
+export default [
+    {
+        // ...other config
+        ignores: ["temp.js", ".config/*"]
+    }
+];
+```
+
+Also, dotfiles (e.g. `.dotfile.js`) are no longer ignored by default with `eslint.config.js`.
+
+### CLI Flag Changes
+
+The following CLI flags are no longer supported with the `eslint.config.js` file format:
+
+* `--rulesdir`
+* `--ext`
+* `--resolve-plugins-relative-to`
+
+The flag `--no-eslintrc` has been replaced with `--no-config-lookup`.
+
+### Additional Changes
+
+The following changes have been made from the `.eslintrc` to the `eslint.config.js` file format:
+
+* The `root` option no longer exists.
+* The `files` option cannot be a single string anymore, it must be an array.
+* The `sourceType` option now supports the new value `"commonjs"` (`.eslintrc` supports it too, but it was never documented).
+* You can configure `noInlineConfig` and `reportUnusedDisableDirectives` options under the setting `linterOptions`.
+
+## Things That Haven’t Changed between Configuration File Formats
+
+While all the above mentioned features have changed from `.eslintrc` to `eslint.config.js` configurations, the following has stayed the same:
+
+* Syntax for adding rules
+* Processors
+* All functionality. Just the way to configure it has changed.
```

Comment:
I'm not sure what "all functionality" means here. I think we can remove this.

Reply:
by this, i mean "how ESLint lints files and reports errors". just the way to set up the configuration has changed. 

i'll play w the phrasing a bit here.